pkg/networking/node_eni_info_resolver.go

```diff
@@ -128,13 +128,13 @@ func (r *defaultNodeENIInfoResolver) resolveViaInstanceID(ctx context.Context, n
 }
 
 // findInstancePrimaryENI returns the primary ENI among list of eni on an EC2 instance
-func findInstancePrimaryENI(enis []*ec2sdk.InstanceNetworkInterface) (*ec2sdk.InstanceNetworkInterface, error) {
+func findInstancePrimaryENI(enis []ec2types.InstanceNetworkInterface) (ec2types.InstanceNetworkInterface, error) {
 	for _, eni := range enis {
-		if awssdk.Int64Value(eni.Attachment.DeviceIndex) == 0 {
+		if awssdk.ToInt32(eni.Attachment.DeviceIndex) == 0 {
 			return eni, nil
 		}
 	}
-	return nil, errors.Errorf("[this should never happen] no primary ENI found")
+	return ec2types.InstanceNetworkInterface{}, errors.Errorf("[this should never happen] no primary ENI found")
```

Comment:
nit: maybe just return nil when there's error

Reply:
Not a pointer type anymore. please refer the comment above.